### PR TITLE
helix: 22.11 -> 23.03

### PR DESCRIPTION
**Pull request by leona-ya**
###### Description of changes

https://github.com/helix-editor/helix/blob/master/CHANGELOG.md#2303-2023-03-31

Superseeds #224062. We must use cargo lock for now, as upstream uses a git checkout of tree-sitter (via `patch.crates-io`)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
@GrahamcOfBorg eval

File: pkgs/applications/editors/helix/default.nix [hunk 1]
```diff
@@ -2,17 +2,25 @@
...
     url = "https://github.com/helix-editor/helix/releases/download/${version}/helix-${version}-source.tar.xz";
-    sha256 = "sha256-En6SOyAPNPPzDGdm2XTjbGG0NQFGBVzjjoyCbdnHFao=";
+    sha256 = "sha256-FtY2V7za3WGeUaC2t2f63CcDUEg9zAS2cGUWI0YeGwk=";
     stripRoot = false;
   };
 
-  cargoSha256 = "sha256-oSS0LkLg2JSRLYoF0+FVQzFUJtFuVKtU2MWYenmFC0s=";
+  # should be removed, when tree-sitter is not used as a git checkout anymore
+  cargoLock = {
+    lockFile = "${src}/Cargo.lock";
```
Comment (mweinelt) [inline pkgs/applications/editors/helix/default.nix:17]:
```suggestion
    lockFile =  ./Cargo.lock;
```

File: pkgs/applications/editors/helix/default.nix [hunk 2]
```diff
@@ -2,17 +2,25 @@
...
+    outputHashes = {
+      "tree-sitter-0.20.9" = "sha256-/PaFaASOT0Z8FpipX5uiRCjnv1kyZtg4B9+TnHA0yTY=";
+    };
+  };
+
+  cargoSha256 = "sha256-+KnBQA7gYLu2O/5vbY5cdEj9hni0Cn3cWPYswBi4934=";
```
Comment (mweinelt) [inline pkgs/applications/editors/helix/default.nix:23]:
```suggestion
  cargoHash = "sha256-+KnBQA7gYLu2O/5vbY5cdEj9hni0Cn3cWPYswBi4934=";
```

File: pkgs/applications/editors/helix/default.nix [hunk 3]
```diff
@@ -2,17 +2,25 @@
...
   pname = "helix";
-  version = "22.12";
+  version = "23.03";
 
   # This release tarball includes source code for the tree-sitter grammars,
   # which is not ordinarily part of the repository.
   src = fetchzip {
     url = "https://github.com/helix-editor/helix/releases/download/${version}/helix-${version}-source.tar.xz";
-    sha256 = "sha256-En6SOyAPNPPzDGdm2XTjbGG0NQFGBVzjjoyCbdnHFao=";
+    sha256 = "sha256-FtY2V7za3WGeUaC2t2f63CcDUEg9zAS2cGUWI0YeGwk=";
```
Comment (mweinelt) [inline pkgs/applications/editors/helix/default.nix:11]:
```suggestion
    hash = "sha256-FtY2V7za3WGeUaC2t2f63CcDUEg9zAS2cGUWI0YeGwk=";
```

**Comment by james-atkins** [comment]
The Cargo.lock file is already in the source tarball we download (and check the hash of) - so there is no need to add it to nixpkgs.

The `cargoHash` attribute is also [unnecessary](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#importing-a-cargolock-file-importing-a-cargolock-file).

**Comment by mweinelt** [comment]
Except `buildins.readFile` does not seem to want to read from `${src}/Cargo.lock`. That's probably [IFD](https://nixos.wiki/wiki/Import_From_Derivation).

File: pkgs/applications/editors/helix/default.nix (hunk 2 above)
Comment (figsoda) [reply pkgs/applications/editors/helix/default.nix:23]: we already have `cargoLock`
```suggestion
```

**Comment by leona-ya** [comment]
Sorry @mweinelt. That was a product of me way to sleepy to do nix. I fixed up the PR.

**Pull request merged**